### PR TITLE
Bugfixes/fix_common_azure_issues

### PR DESCRIPTION
**Pull request by irmatov**
* fix nil pointer dereference in network public ip addresses
* add missing value handling case in security settings

Closes cloudquery/cloudquery-issues#115

### CODE REVIEW

File: resources/monitor_activity_logs.go [hunk 1]
```diff
@@ -244,8 +244,16 @@ func fetchMonitorActivityLogs(ctx context.Context, meta schema.ClientMeta, paren
 	if err != nil {
 		return err
 	}
+	// azure returns same events sometimes so we have to filter out the duplicates
+	seen := make(map[string]struct{})
```
Comment (roneli) [inline resources/monitor_activity_logs.go:248]: why not map[string]bool ?

File: resources/monitor_diagnostic_settings.go [hunk 2]
```diff
@@ -13,11 +13,12 @@ import (
...
-		Name:        "azure_monitor_diagnostic_settings",
-		Description: "DiagnosticSettingsResource the diagnostic setting resource",
-		Resolver:    fetchMonitorDiagnosticSettings,
-		Multiplex:   client.SubscriptionMultiplex,
-		Options:     schema.TableCreationOptions{PrimaryKeys: []string{"subscription_id", "id"}},
+		Name:         "azure_monitor_diagnostic_settings",
+		Description:  "DiagnosticSettingsResource the diagnostic setting resource",
+		Resolver:     fetchMonitorDiagnosticSettings,
+		Multiplex:    client.SubscriptionMultiplex,
+		DeleteFilter: client.DeleteSubscriptionFilter,
```
Comment (roneli) [inline resources/monitor_diagnostic_settings.go:20]: I would just double check all resources that they have a deletefilter defined.

**Review by roneli — CHANGES_REQUESTED** [review]
LGTM, a few changes suggested.

**Comment by roneli** [comment]
I think for cloudquery/cloudquery-issues#115 rbac issues this requires the user to add special permissions which is quite tricky in azure-ad.  I would maybe add a special IgnoreError for those resources.

File: resources/monitor_activity_logs.go (hunk 1 above)
Comment (irmatov) [reply resources/monitor_activity_logs.go:248]: It doesn't matter here, but map[string]struct{} consumes less memory than map[string]bool if all we want to test if an key is present in map or not.

File: resources/monitor_diagnostic_settings.go (hunk 2 above)
Comment (irmatov) [reply resources/monitor_diagnostic_settings.go:20]: Good point. I have checked all resources and found a few more without a DeleteFilter. Thanks.

**Comment by irmatov** [comment]
I am not sure if ignoring an error is the way to go here. If a user fetches those resources and it fails it should be an error. If a user doesn't want to configure proper access then he can just comment out (disable) those resources.

**Review by roneli — APPROVED** [review]
LGTM

**Pull request merged**